frontend/packages/kubevirt-plugin/src/components/vms/pending-changes-warning.tsx

```diff
@@ -0,0 +1,200 @@
+import { vmFlavorModal } from '../modals';
+import { VMCDRomModal } from '../modals/cdrom-vm-modal/vm-cdrom-modal';
+import { BootOrderModal } from '../modals/boot-order-modal';
+import { PendingChangesAlert } from '../Alerts/PendingChangesAlert';
+import {
+  List,
+  ListItem,
+  Button,
+  Breadcrumb,
+  BreadcrumbItem,
+  BreadcrumbHeading,
+  ButtonVariant,
+} from '@patternfly/react-core';
+import { asVM } from '../../selectors/vm';
+import { history } from '@console/internal/components/utils/router';
+import { getNamespace, getName } from '@console/shared';
+import { VirtualMachineModel, VirtualMachineInstanceModel } from '../../models';
+
+import './pending-changes-warning.scss';
+import { PENDING_CHANGES_WARNING_MESSAGE } from '../../strings/vm/status';
+import { isVMRunningOrExpectedRunning } from '../../selectors/vm/selectors';
+
+type PendingChangesWarningProps = {
+  vmi?: FirehoseResult<VMIKind>;
+  vmLikeEntity?: FirehoseResult<VMLikeEntityKind>;
+};
+
+const getPendingChangesByTab = (pendingChanges: PendingChanges) => {
+  const pendingChangesByTab = {
+    [VMTabEnum.details]: [],
+    [VMTabEnum.env]: [],
+    [VMTabEnum.nics]: [],
+    [VMTabEnum.disks]: [],
+  };
+
+  Object.keys(pendingChanges).forEach((key) => {
+    const pc = pendingChanges[key];
+    pc.isPendingChange && pendingChangesByTab[pc.vmTab].push(key);
+  });
+
+  return pendingChangesByTab;
+};
+
+type WarningTabRowProps = {
+  tabName: string;
+  tabProps: string[];
+  pendingChanges: PendingChanges;
+};
+
+const WarningTabRow: React.FC<WarningTabRowProps> = ({ tabName, tabProps, pendingChanges }) => (
+  <Breadcrumb className="kv-warning--pf-c-breadcrumb">
+    <BreadcrumbHeading key={`${tabName}-header-${tabProps.join('-')}`}>{tabName}</BreadcrumbHeading>
+    <BreadcrumbItem key={`${tabName}-${tabProps.join('-')}`}>
+      {tabProps.map((key, idx) => (
+        <div key={`${tabName}-${key}`} className="kv-warning-changed-attrs">
+          <Button isInline onClick={pendingChanges[key].execAction} variant={ButtonVariant.link}>
+            {key}
+          </Button>
+          <div className="kv-warning--comma-margin">{idx === tabProps.length - 1 ? '' : ','}</div>
+        </div>
+      ))}
+    </BreadcrumbItem>
+  </Breadcrumb>
+);
+
+export const getPendingChanges = (
+  vm: VMKind,
+  vmWrapper: VMWrapper,
+  vmiWrapper: VMIWrapper,
+): PendingChanges => ({
+  [IsPendingChange.flavor]: {
+    isPendingChange: isFlavorChanged(vmWrapper, vmiWrapper),
+    execAction: () => vmFlavorModal({ vmLike: vm, blocking: true }),
+    vmTab: VMTabEnum.details,
+  },
+  [IsPendingChange.cdroms]: {
+    isPendingChange: isCDROMChanged(vmWrapper, vmiWrapper),
+    execAction: () => VMCDRomModal({ vmLikeEntity: vm, modalClassName: 'modal-lg' }),
+    vmTab: VMTabEnum.details,
+  },
+  [IsPendingChange.bootOrder]: {
+    isPendingChange: isBootOrderChanged(vmWrapper, vmiWrapper),
+    execAction: () => BootOrderModal({ vmLikeEntity: vm, modalClassName: 'modal-lg' }),
```

Comment:
this opens the modal at any page. Maybe it would be better to have it open in details page

Reply:
plus other modals as well

Reply:
the idea is to open then modal, I can redirect to the details page first and then open the modal if that make more sense

Reply:
+1 that would be nice